services/product-service/tests/test_config.py

```
@@ -0,0 +1,15 @@
+from pydantic import ValidationError
+
+
+def test_internal_service_token_rejects_empty_string() -> None:
+    """INTERNAL_SERVICE_TOKEN must fail at settings load time when empty."""
+    with pytest.raises(ValidationError):
+        Settings(internal_service_token="")
+
+
+def test_internal_service_token_accepts_non_empty_value() -> None:
+    """A non-empty internal service token should load normally."""
+    settings = Settings(internal_service_token="test-token")
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to argument: "internal_service_token"

(S106)